### ISSUE TITLE
Error occurs with treemacs-icon-dired-mode when running dired for the first time.

### ISSUE THREAD

**Issue by taquangtrung** [Alexander-Miller/treemacs#668]
Hi,

When I run dired-mode for the first time after opening Emacs, the following of treemacs-icon-dired-mode will happen:

```
treemacs-icons-dired--display: Wrong type argument: char-or-string-p, nil
```

Here is the stack trace in my Emacs:

```
Debugger entered--Lisp error: (wrong-type-argument char-or-string-p nil)
  treemacs-icons-dired--display-icons-for-subdir("/home/trungtq/.emacs.d/private/" #<marker at 1 in private>)
  treemacs-icons-dired--display()
  run-hooks(dired-after-readin-hook)
  dired-readin()
  dired-internal-noselect("~/.emacs.d/private/" nil)
  dired-noselect("/home/trungtq/.emacs.d/private/" nil)
  #f(compiled-function (dirname &optional switches) "\"Edit\" directory DIRNAME--delete, rename, print, etc. some files in it.\nOptional second argument SWITCHES specifies the `ls' options used.\n(Interactively, use a prefix argument to be able to specify SWITCHES.)\n\nIf DIRNAME is a string, Dired displays a list of files in DIRNAME (which\nmay also have shell wildcards appended to select certain files).\n\nIf DIRNAME is a cons, its first element is taken as the directory name\nand the rest as an explicit list of files to make directory entries for.\nIn this case, SWITCHES are applied to each of the files separately, and\ntherefore switches that control the order of the files in the produced\nlisting have no effect.\n\n\\<dired-mode-map>You can flag files for deletion with \\[dired-flag-file-deletion] and then\ndelete them by typing \\[dired-do-flagged-delete].\nType \\[describe-mode] after entering Dired for more info.\n\nIf DIRNAME is already in a Dired buffer, that buffer is used without refresh." (interactive (dired-read-dir-and-switches "")) #<bytecode 0x2484bed>)("/home/trungtq/.emacs.d/private/" nil)
  ad-Advice-dired(#f(compiled-function (dirname &optional switches) "\"Edit\" directory DIRNAME--delete, rename, print, etc. some files in it.\nOptional second argument SWITCHES specifies the `ls' options used.\n(Interactively, use a prefix argument to be able to specify SWITCHES.)\n\nIf DIRNAME is a string, Dired displays a list of files in DIRNAME (which\nmay also have shell wildcards appended to select certain files).\n\nIf DIRNAME is a cons, its first element is taken as the directory name\nand the rest as an explicit list of files to make directory entries for.\nIn this case, SWITCHES are applied to each of the files separately, and\ntherefore switches that control the order of the files in the produced\nlisting have no effect.\n\n\\<dired-mode-map>You can flag files for deletion with \\[dired-flag-file-deletion] and then\ndelete them by typing \\[dired-do-flagged-delete].\nType \\[describe-mode] after entering Dired for more info.\n\nIf DIRNAME is already in a Dired buffer, that buffer is used without refresh." (interactive (dired-read-dir-and-switches "")) #<bytecode 0x2484bed>) "/home/trungtq/.emacs.d/private/")
  apply(ad-Advice-dired #f(compiled-function (dirname &optional switches) "\"Edit\" directory DIRNAME--delete, rename, print, etc. some files in it.\nOptional second argument SWITCHES specifies the `ls' options used.\n(Interactively, use a prefix argument to be able to specify SWITCHES.)\n\nIf DIRNAME is a string, Dired displays a list of files in DIRNAME (which\nmay also have shell wildcards appended to select certain files).\n\nIf DIRNAME is a cons, its first element is taken as the directory name\nand the rest as an explicit list of files to make directory entries for.\nIn this case, SWITCHES are applied to each of the files separately, and\ntherefore switches that control the order of the files in the produced\nlisting have no effect.\n\n\\<dired-mode-map>You can flag files for deletion with \\[dired-flag-file-deletion] and then\ndelete them by typing \\[dired-do-flagged-delete].\nType \\[describe-mode] after entering Dired for more info.\n\nIf DIRNAME is already in a Dired buffer, that buffer is used without refresh." (interactive (dired-read-dir-and-switches "")) #<bytecode 0x2484bed>) "/home/trungtq/.emacs.d/private/")
  dired("/home/trungtq/.emacs.d/private/")
  ido-file-internal(dired dired nil "Dired: " dir)
  ido-dired()
  funcall-interactively(ido-dired)
  call-interactively(ido-dired nil nil)
  command-execute(ido-dired)
```

Could you help to check if this is an issue of Treemacs or it is due to my Emacs settings?

Thank you!

**Comment by Alexander-Miller** [comment]
Do you use spacemacs? If yes, then it's been fixed recently: https://github.com/Alexander-Miller/treemacs/issues/480

**Comment by taquangtrung** [comment]
Yes, I'm using Spacemac. I already updated Treemacs to the newest version (before reporting the bug), and the bug still occurs.

**Comment by Alexander-Miller** [comment]
It's spacemacs you need to update.

**Comment by taquangtrung** [comment]
For some reason, I pulled the newest code in the develop branch of Spacemacs and `treemacs-icon-dired-mode` is disable by default. When I manually enable it, the error still happens. 

Here is the newest commit in my Spacemacs. Perhaps it is still problematic. Hope that it will be fixed in Spacemacs soon.

```
develop origin/develop d0f5477c323ef6600e29d22b75d876c49916b3f4
Author:     emacspace <emacspace@gmail.com>
AuthorDate: Wed May 6 22:11:32 2020 +0000
Commit:       Eugene Yaremenko <w3techplayground@gmail.com>
CommitDate:   Thu May 7 01:52:26 2020 +0300
```

**Comment by Alexander-Miller** [comment]
I cannot reproduce that, at least for the most part. I just ran a clean install of the latest develop branch, and the mode was not enabled on the first start, when spacemacs first had to install all its packages. After restarting calling `dired` would show icons as expected.

Did you try things multiple times, after restarting spacemacs? And how exactly do you start dired?

**Comment by Alexander-Miller** [comment]
Just encountered the same error when dired-icons mode is manually turned off and on again, investigating now.

**Comment by Alexander-Miller** [comment]
Bug should be fixed now.

**Comment by taquangtrung** [comment]
Great! I tested again, and it works perfectly fine.

Just a minor question. I remember previously that `treemacs-icon-dired-mode` is enabled by default. But after pulling the newest code, I have to manually enable it. Is this a new and intended behavior?

**Comment by Alexander-Miller** [comment]
No, it should be enabled by default. See
 https://github.com/syl20bnr/spacemacs/blob/develop/layers/+filetree/treemacs/config.el#L40
and 
https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bfiletree/treemacs/packages.el#L100

So unless you somehow prevent dired-load-hook from running it you should see the icons whenever you start dired.

**Comment by taquangtrung** [comment]
Thanks for the suggestion. I didn't interfere with `dired-load-hook`. Its value is still `(treemacs-icons-dired-mode)` as you said.

Besides, here is the value of `dired-mode-hook`  (where `hook-dired-omit` is my function, and the rests belong to Spacemacs)

```
(hook-dired-omit diredp-nb-marked-in-mode-name diredp--set-up-font-locking
dired-extra-startup spacemacs//init-jump-handlers-dired-mode)
```

I'm currently investigating what happens in my Emacs.

**Comment by taquangtrung** [comment]
Hi,

I finally realized that the hook that I added to `dired-mode-hook` to enable `dired-omit-mode` prevents `treemacs-icons-dired-mode` from being loaded.

Perhaps there are conflicts between the two hooks or between `dired-omit` and `treemacs`. If I remove my hook, than `treemacs-icon-dired-mode` will work perfectly fine.

Thanks again for your contributions to this awesome package `treemacs`! :)
